packages/plugins/src/components/plugin-area/index.js

```diff
@@ -50,80 +56,61 @@ import { getPlugins } from '../../api';
  *
  * @return {WPComponent} The component to be rendered.
  */
-class PluginArea extends Component {
-	constructor() {
-		super( ...arguments );
-
-		this.setPlugins = this.setPlugins.bind( this );
-		this.memoizedContext = memoize( ( name, icon ) => {
-			return {
-				name,
-				icon,
-			};
-		} );
-		this.state = this.getCurrentPluginsState();
-	}
+function PluginArea( { scope, onError } ) {
+	const store = useMemo( () => {
+		let lastValue;
 
-	getCurrentPluginsState() {
 		return {
-			plugins: getPlugins( this.props.scope ).map(
-				( { icon, name, render } ) => {
-					return {
-						Plugin: render,
-						context: this.memoizedContext( name, icon ),
-					};
-				}
-			),
-		};
-	}
+			subscribe( listener ) {
+				addAction(
+					'plugins.pluginRegistered',
+					'core/plugins/plugin-area/plugins-registered',
+					listener
+				);
+				addAction(
+					'plugins.pluginUnregistered',
+					'core/plugins/plugin-area/plugins-unregistered',
+					listener
+				);
+				return () => {
+					removeAction(
+						'plugins.pluginRegistered',
+						'core/plugins/plugin-area/plugins-registered'
+					);
+					removeAction(
+						'plugins.pluginUnregistered',
+						'core/plugins/plugin-area/plugins-unregistered'
+					);
+				};
+			},
+			getValue() {
+				const nextValue = getPlugins( scope );
 
-	componentDidMount() {
-		addAction(
-			'plugins.pluginRegistered',
-			'core/plugins/plugin-area/plugins-registered',
-			this.setPlugins
-		);
-		addAction(
-			'plugins.pluginUnregistered',
-			'core/plugins/plugin-area/plugins-unregistered',
-			this.setPlugins
-		);
-	}
+				if ( ! isShallowEqual( lastValue, nextValue ) ) {
+					lastValue = nextValue;
+				}
 
-	componentWillUnmount() {
-		removeAction(
-			'plugins.pluginRegistered',
-			'core/plugins/plugin-area/plugins-registered'
-		);
-		removeAction(
-			'plugins.pluginUnregistered',
-			'core/plugins/plugin-area/plugins-unregistered'
-		);
-	}
+				return lastValue;
+			},
+		};
+	}, [ scope ] );
 
-	setPlugins() {
-		this.setState( this.getCurrentPluginsState );
-	}
+	const plugins = useSyncExternalStore( store.subscribe, store.getValue );
```

Comment:
I think `plugins` and `store` could be abstracted to their own `usePluginsState` hook. Not a huge deal of course, since it's used only in one place.

Reply:
I also thought about it. Let’s abstract when there’s another use for them.